tests/functional/aws-node-sdk/test/object/100-continue.js

```diff
@@ -0,0 +1,123 @@
+import assert from 'assert';
+import http from 'http';
+import url from 'url';
+
+import withV4 from '../support/withV4';
+import BucketUtility from '../../lib/utility/bucket-util';
+
+const bucket = 'foo-bucket';
+const key = 'foo-key';
+const body = Buffer.alloc(1024 * 1024);
+
+class ContinueRequestHandler {
+    constructor(path) {
+        this.path = path;
+        return this;
+    }
+
+    setRequestPath(path) {
+        this.path = path;
+        return this;
+    }
+
+    setExpectHeader(header) {
+        this.expectHeader = header;
+        return this;
+    }
+
+    getRequestOptions() {
+        return {
+            path: this.path,
+            hostname: 'localhost',
+            port: 8000,
+            method: 'PUT',
+            headers: {
+                'content-length': body.length,
+                'Expect': this.expectHeader || '100-continue',
+            },
+        };
+    }
+
+    hasStatusCode(statusCode, cb) {
+        const options = this.getRequestOptions();
+        const req = http.request(options, res => {
+            assert.strictEqual(res.statusCode, statusCode);
+            return cb();
+        });
+        // Send the body either on the continue event, or immediately.
+        if (this.expectHeader === '100-continue') {
+            req.on('continue', () => req.end(body));
+        } else {
+            req.end(body);
+        }
+    }
+
+    sendsBodyOnContinue(cb) {
+        const options = this.getRequestOptions();
+        const req = http.request(options);
+        // At this point we have only sent the header.
+        assert(req.output.length === 1);
+        const headerLen = req.output[0].length;
+        req.on('continue', () => {
+            // Has only the header been sent?
+            assert.strictEqual(req.socket.bytesWritten, headerLen);
+            // Send the body since the continue event has been emitted.
+            return req.end(body);
+        });
+        req.on('close', () => {
+            const expected = body.length + headerLen;
+            // Has the entire body been sent?
+            assert.strictEqual(req.socket.bytesWritten, expected);
+            return cb();
+        });
+        req.on('error', err => cb(err));
+    }
+}
+
+describe('PUT public object with 100-continue header', () => {
```

Comment:
It would be good to test the streaming v4 cases to make sure it works.  You can mess with either the java or the fog tests to add a 100-continue header and make sure the tests still pass.